includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -172,7 +172,9 @@ public function handle_promote_job() {
 		if ( ! $post_id ) {
 			wp_die( esc_html__( 'No job listing ID provided for promotion.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
 		}
-		if ( ! $this->can_promote_job( $post_id ) ) {
+
+		$is_promoted_or_published = WP_Job_Manager_Promoted_Jobs::is_promoted( $post_id ) || 'publish' === get_post_status( $post_id );
```

Comment:
It was like this before. I extracted the `publish` check because it depends on the context now the changes of this PR. But if you have a good idea of how to refactor it, feel free to comment here so that I can change it. 😉

Reply:
Still I think that having this 'publish' status checks everywhere might not be a good idea.

What about:
- Rename the `$this->can_promote_job( $post_id )` to `$this->can_current_user_promote_job( $post_id )`.
- Add a method `WP_Job_Manager_Promoted_Jobs::can_be_promoted( $post_id )` encapsulating the 'publish' status check? This method description must be something like "Check if a post_id can be applied the promote action or not". And would be used here and in the elseif to show the promote button.

If you think it's a bit confusing or that it might not be worth the effort, feel free to continue with what you have :)

Reply:
Have you considered the edit when you thought about this?
I tried to refactor it as you said, but I couldn't achieve good results. The reason is that if we had a separate `can_be_promoted`, it would duplicate the check of `can_current_user_promote_job`, which we also want to check previously.

Anyway, I changed this part a little to improve the readability. It made this part a little bigger, but I think it was worth it. https://github.com/Automattic/WP-Job-Manager/pull/2544/commits/a68e011fe551846407d101fd218d58174c8d4248

WDYT?